src/marqo/s2_inference/clip_utils.py

```diff
@@ -348,7 +363,7 @@ def load(self) -> None:
                 self.model_path = self._download_from_repo()
             elif os.path.isfile(path):
                 self.model_path = path
-            elif validators.url(path):
+            elif validate_url(path):
                 self.model_path = download_model(url=path)
```

Comment:
we don't encode url here. So the validation can pass but the download can fail. (if download_mode cannot not handle unencoded characters in the url)